module-2/backend/week_two.md

```diff
@@ -4,35 +4,112 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-1. At a high level, what is ActiveRecord? What does it do/allow you to do?
-2. What is a migration?
-3. How does a table relate to a model?
-4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
-5. What do they allow you to do?
-6. What's the difference between agile workflow and waterfall method?
-7. What is the difference between `#new` and `#create`?
+1. At a high level, what is ActiveRecord? What does it do/allow you to do?<br>
+
+ActiveRecord is a piece of rails that can be used independently. It does Object Relational Mapping which interacts with a database through models (which are our objects) that go between the database and the views. It provides a ton of built in functionality to make our lives easier, based on the idea that convention should be triumphed over configuration. 
+
+2. What is a migration?<br>
+
+A migration is a file/set of instructions for building the database. In the migration files we say which tables we want with what columns and additionally we can change something that has already been created i.e. rename columns, add columns, delete, etc.
+
+3. How does a table relate to a model?<br>
+
+Due to the rails "magic", a model looks at the table and assumes that we would like the model object to have attributes that correspond to all the columns in a given table. A table is the database and the model is built off of it. The model seems to be our way of interacting with the database also. We can create relationships between tables and thus models.
+
+4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.<br>
+
+I would say that these relational methods are instance methods. An article belongs to its author, and if we had the whole thing set up we could assign a particular author to a variable and then call articles on it i.e. `author = Author.first` & `author.articles` and it should bring up all the articles belonging to that particular author (compared with calling Article.all). 
```

Comment:
I think you're getting the bigger point, which is that the methods that `belongs_to` and `has_many` offers us are instance methods. So, for example:

```ruby
blue_car = Car.find(1) # Gives us a car with an ID of 1.
ford_dealership = blue_car.dealer # assigns the Dealer associated with the `blue_car` to the variable `ford_dealership`
```